homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -0,0 +1,204 @@
+        nearby_stations = data["stations"]
+        if not nearby_stations:
+            _LOGGER.warning(
+                "Could not find any station in range. Try with a bigger radius"
+            )
+        else:
+            for station in data["stations"]:
+                self.add_station(station)
+
+        # Add manually specified additional stations
+        for station_id in additional_stations:
+            try:
+                additional_station_data = pytankerkoenig.getStationData(
+                    self._api_key, station_id
+                )
+            except pytankerkoenig.customException as err:
+                additional_station_data = {
+                    "ok": False,
+                    "message": err,
+                    "exception": True,
+                }
+
+            if not additional_station_data["ok"]:
+                _LOGGER.error(
+                    "Error when adding station %s:\n %s",
+                    station_id,
+                    additional_station_data["message"],
+                )
+                # Clear entity dictionary, so that the platform setup fails / no sensors get loaded in hass
+                self._entities = {}
+            else:
+                self.add_station(additional_station_data["station"])
+
+    @property
+    def entity_list(self):
+        """Get the list of all entities the platform is monitoring."""
+
+        # Get flat list from the dictionary of lists
+        return [
+            sensor_entity
+            for sublist in self._entities.values()
+            for sensor_entity in sublist
+        ]
+
+    def update(self, now=None):
+        """Get the latest data from tankerkoenig.de."""
+        _LOGGER.debug("Fetching new data from tankerkoenig.de")
+        entity_list = list(self._entities.keys())
+        data = pytankerkoenig.getPriceList(self._api_key, entity_list)
+
+        if data["ok"]:
+            _LOGGER.debug("Received data: %s", data)
+            for station_id, station_list in self._entities.items():
+                for station in station_list:
+                    station.new_data(data["prices"].get(station_id))
+        else:
+            _LOGGER.error(
+                "Error fetching data from tankerkoenig.de: %s", data["message"]
+            )
+
+    def add_station(self, station: dict):
+        """Add fuel station to the entity list."""
+        station_id = station["id"]
+        if station_id in self._entities.keys():
+            _LOGGER.warning(
+                "Sensor for station with id %s was already created", station_id
+            )
+            return
+
+        self._entities[station_id] = []
+        _LOGGER.debug(
+            "add_station called for station: %s and fuel types: %s",
+            station,
+            self._fuel_types,
+        )
+        for fuel in self._fuel_types:
+            if fuel in station.keys():
+                station_sensor = FuelPriceSensor(
```

Comment:
The entity is a platform concern and should be contained within its platform. The entity should be created and added within its platform.